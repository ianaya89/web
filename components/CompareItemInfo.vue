<template>
  <div class="compare-item">
    <inventory-item-info :title="title" :info="info" :src="src">
      <template v-slot:image>
        <slot name="image" />
      </template>
    </inventory-item-info>
    <div v-if="remover" class="flex pl-4">
      <span class="compare-item-remover text-p4" :class="[{'lni-close': !removing},{'fas fa-spin fa-spinner':removing}]" @click.stop="onClick()" />
    </div>
  </div>
</template>

<script>
import InventoryItemInfo from '@/components/inventory/InventoryItemInfo'

export default {
  name: 'CompareItemInfo',

  components: {
    InventoryItemInfo
  },

  props: {
    id: {
      type: String,
      required: true
    },
    title: {
      type: String,
      required: true
    },
    src: {
      type: String,
      default: require('../assets/images/thumb-fallback.svg')
    },
    info: {
      type: String,
      default: null
    },
    remover: {
      type: Boolean,
      default: false
    }
  },

  data: () => ({
    removing: false
  }),

  methods: {
    onClick() {
      this.removing = true
      this.$emit('compare-item-remove', this.id)
      this.$store.dispatch('compare/remove', this.id)
    }
  }
}
</script>

<style lang="postcss" scoped>
.compare-item {
  @apply .w-full .flex .py-4;

  .compare-item-remover {
    @apply .cursor-pointer .m-auto .text-grey-l2;
    transition: color 150ms ease-in-out;
  }

  &:hover .compare-item-remover {
    @apply .text-red;
  }
}
</style>
