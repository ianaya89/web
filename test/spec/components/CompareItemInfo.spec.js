import { shallowMount } from '@vue/test-utils'
import CompareItemInfo from '@/components/CompareItemInfo'

describe('CompareItemInfo', () => {
  it('renders CompareItemInfo value 3', () => {
    const wrapper = shallowMount(CompareItemInfo, {
      propsData: {
        id: '3',
        title: 'Title',
        info: 'Small info',
        src: 'someAvatarSource'
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('renders CompareItemInfo and remove it', () => {
    const wrapper = shallowMount(CompareItemInfo, {
      propsData: {
        id: '3',
        title: 'Title',
        info: 'Small info',
        src: 'someAvatarSource',
        remover: true
      }
    })

    expect(wrapper.element).toMatchSnapshot()
    const remover = wrapper.find('.compare-item-remover')
    remover.trigger('click')
    expect(wrapper.emitted('compare-item-remove')).toBeTruthy()
  })
})
