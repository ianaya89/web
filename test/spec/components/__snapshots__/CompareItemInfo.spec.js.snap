// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareItemInfo renders CompareItemInfo and remove it 1`] = `
<div
  class="compare-item"
>
  <inventory-item-info-stub
    classes="h-10 w-10"
    info="Small info"
    src="someAvatarSource"
    title="Title"
  />
   
  <div
    class="flex pl-4"
  >
    <span
      class="compare-item-remover text-p4 lni-close"
    />
  </div>
</div>
`;

exports[`CompareItemInfo renders CompareItemInfo value 3 1`] = `
<div
  class="compare-item"
>
  <inventory-item-info-stub
    classes="h-10 w-10"
    info="Small info"
    src="someAvatarSource"
    title="Title"
  />
   
  <!---->
</div>
`;
